### PR TITLE
CPM-603: Use UpsertProductCommand in Enrichment E2E tests

### PR DESCRIPTION
**Pull request by cemorin**
Use UpsertProductCommand in Enrichment E2E tests and its related user intents

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php [hunk 1]
```diff
@@ -1198,15 +1198,20 @@ private function updateProduct(UserIntent $userIntent): void
...
-        $this->get('pim_catalog.updater.product')->update($product, ['values' => $values]);
-        $errors = $this->get('pim_catalog.validator.product')->validate($product);
-        if (0 !== $errors->count()) {
-            throw new \Exception(sprintf('Impossible to setup test in %s: %s', static::class, $errors->get(0)->getMessage()));
-        }
-        $this->get('pim_catalog.saver.product')->save($product);
+        $command = UpsertProductCommand::createFromCollection(
+            userId: $this->getUserId('admin'),
+            productIdentifier: $identifier,
+            userIntents: $userIntents
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php:1209]: The family is forgotten, we have to add the SetFamily() in the user intent list

File: tests/back/Pim/Enrichment/EndToEnd/InternalApiTestCase.php [hunk 2]
```diff
@@ -39,25 +42,22 @@ protected function authenticate(UserInterface $user): void
...
-                static::class,
-                $errors->get(0)->getMessage()
-            ));
-        }
-
-        $this->get('pim_catalog.saver.product')->save($product);
+        $command = UpsertProductCommand::createFromCollection(
+            userId: $this->getUserId('admin'),
+            productIdentifier: $identifier,
+            userIntents: $userIntents
```
Comment (nmarniesse) [inline tests/back/Pim/Enrichment/EndToEnd/InternalApiTestCase.php:53]: Add the family please

File: tests/back/Pim/Enrichment/EndToEnd/Product/Product/ExternalApi/AbstractProductTestCase.php [hunk 3]
```diff
@@ -17,59 +19,39 @@
...
-        }
-
-        $this->get('pim_catalog.saver.product')->save($product);
+        $command = UpsertProductCommand::createFromCollection(
+            userId: $this->getUserId('admin'),
+            productIdentifier: $identifier,
+            userIntents: $userIntents
+        );
+        $this->get('pim_enrich.product.message_bus')->dispatch($command);
+        $this->getContainer()->get('pim_catalog.validator.unique_value_set')->reset();
```
Comment (nmarniesse) [inline tests/back/Pim/Enrichment/EndToEnd/Product/Product/ExternalApi/AbstractProductTestCase.php:32]: To remove

File: tests/back/Pim/Enrichment/EndToEnd/Product/Product/ExternalApi/Media/CreateProductMediaFileEndToEnd.php [hunk 4]
```diff
@@ -308,8 +310,18 @@ protected function setUp(): void
...
-        $this->get('pim_catalog.saver.product')->save($product);
+        $command = UpsertProductCommand::createFromCollection(
+            userId: $this->getUserId('admin'),
+            productIdentifier: 'foo',
+            userIntents: []
+        );
+        $this->get('pim_enrich.product.message_bus')->dispatch($command);
+        $this->getContainer()->get('pim_catalog.validator.unique_value_set')->reset();
+        $this->get('akeneo_elasticsearch.client.product_and_product_model')->refreshIndex();
+        $this->get('pim_connector.doctrine.cache_clearer')->clear();
```
Comment (nmarniesse) [inline tests/back/Pim/Enrichment/EndToEnd/Product/Product/ExternalApi/Media/CreateProductMediaFileEndToEnd.php:321]: useless?

File: tests/back/Pim/Enrichment/EndToEnd/Product/Product/InternalApi/QuantifiedAssociations/AddQuantifiedAssociationsToProductEndToEnd.php [hunk 5]
```diff
@@ -15,15 +16,7 @@ public function it_add_quantified_associations_to_a_product(): void
...
-                'values' => [
-                    'sku' => [
-                        [
-                            'scope' => null,
-                            'locale' => null,
-                            'data' => 'yellow_chair',
-                        ],
-                    ],
-                ],
+                new SetTextValue('sku', null, null, 'yellow_chair')
```
Comment (nmarniesse) [inline tests/back/Pim/Enrichment/EndToEnd/Product/Product/InternalApi/QuantifiedAssociations/AddQuantifiedAssociationsToProductEndToEnd.php:19]:
```suggestion
                new SetIdentifierValue('sku', 'yellow_chair')
```

**Review by nmarniesse — APPROVED** [review]
GTM after fix

File: src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php [hunk 6]
```diff
@@ -1198,15 +1198,20 @@ private function updateProduct(UserIntent $userIntent): void
...
+        );
+        $this->get('pim_enrich.product.message_bus')->dispatch($command);
+        $this->getContainer()->get('pim_catalog.validator.unique_value_set')->reset();
+        $this->get('akeneo_elasticsearch.client.product_and_product_model')->refreshIndex();
+        $this->clearDoctrineUoW();
```
[the remark this answers is not in the document]
Comment (cemorin) [reply src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php:1214]: Remove the `clearDoctrineUoW` or the `refreshIndex`?

[the remark this answers is not in the document]
Comment (nmarniesse) [reply src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php:1217]: The 3 lines

```
        $this->getContainer()->get('pim_catalog.validator.unique_value_set')->reset();
        $this->get('akeneo_elasticsearch.client.product_and_product_model')->refreshIndex();
        $this->clearDoctrineUoW();
```

**Review by amigaire — APPROVED** [review]

**Pull request merged**